File: 100. Same Tree/isSameTree.js
// Definition for a binary tree node.
class TreeNode {
  constructor(val = 0, left = null, right = null) {
    this.val = val;
    this.left = left;
    this.right = right;
  }
}

/**
 * @param {TreeNode} p
 * @param {TreeNode} q
 * @return {boolean}
 */
const isSameTree = (p, q) => {
  if (!p && !q) {
    return true;
  }

  if (!p || !q) {
    return false;
  }

  if (p.val !== q.val) {
    return false;
  }

  return (
    isSameTree(p.left, q.left) &&
    isSameTree(p.right, q.right)
  )
};

// p = [1, 2, 3], q = [1, 2, 3]
const pTree = new TreeNode(1);
pTree.left = new TreeNode(2);
pTree.right = new TreeNode(3);

const qTree = new TreeNode(1);
qTree.left = new TreeNode(2);
qTree.right = new TreeNode(3);

console.log('qTree: ', isSameTree(pTree, qTree));

File: 100. Same Tree/isSameTree.py
from typing import Optional
# Definition for a binary tree node.
class TreeNode:
  def __init__(self, val=0, left=None, right=None):
    self.val = val
    self.left = left
    self.right = right

class Solution:
  def isSameTree(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:
    if not p and not q:
      return True
      
    if not p or not q:
      return False
      
    if p.val != q.val:
      return False
      
    return (self.isSameTree(p.left, q.left) and
      self.isSameTree(p.right, q.right))
  
pTree = TreeNode(1)
pTree.left = TreeNode(2)
pTree.right = TreeNode(3)

qTree = TreeNode(1)
qTree.left = TreeNode(2)
qTree.right = TreeNode(3)

solution = Solution()
print('isSameTree: ', solution.isSameTree(pTree, qTree))